libbeat/outputs/otelconsumer/otelconsumer.go

```diff
@@ -97,10 +97,24 @@ func (out *otelConsumer) logsPublish(_ context.Context, batch publisher.Batch) e
 		pcommonEvent.CopyTo(logRecord.Body().SetEmptyMap())
 	}
 
-	if err := out.logsConsumer.ConsumeLogs(context.TODO(), pLogs); err != nil {
-		return fmt.Errorf("error otel log consumer: %w", err)
+	err := out.logsConsumer.ConsumeLogs(ctx, pLogs)
+	if err != nil {
+		// Permanent errors shouldn't be retried. This tipically means
+		// the data cannot be serialized by the exporter that is attached
+		// to the pipeline or when the destination refuses the data because
+		// it cannot decode it. Retrying in this case is useless.
+		//
+		// See https://github.com/open-telemetry/opentelemetry-collector/blob/1c47d89/receiver/doc.go#L23-L40
+		if consumererror.IsPermanent(err) {
+			batch.Drop()
+		} else {
+			batch.Retry()
```

Comment:
I think we need to increment `st.RetryableErrors` here as well.

Reply:
Thanks Lee!